### ISSUE TITLE
JetBrains: Crash occurred while adding Enterprise account.

### ISSUE THREAD

**Issue by jay-fibi** [sourcegraph/jetbrains#1444]
### Cody Version

v5.5.9-nightly

### IDE Information

PyCharm 2023.3.4 (Community Edition)
Build #PC-233.14475.56, built on February 26, 2024
Runtime version: 17.0.10+1-b1087.17 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 12
Registry:
  ide.experimental.ui=true
Non-Bundled Plugins:
  com.sourcegraph.jetbrains (5.5.9-nightly)

### Describe the bug

Platform: Windows

**Steps to reproduce:**

1. Have Cody plugin installed in your JB PyCharm editor
2. Login to Cody with PRO user
3. Click on Cody icon from status bar
4. Click on "Manage Accounts" 
5. Click on "+" to add account and choose "Log In with Token to Sourcegraph Enterprise"
6. Add Server and Token and hit "Add Account"

Observed behavior:
Cody crashes with following stracktrace,
Stacktrace:
```text
com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Read access is allowed from inside read-action (see Application.runReadAction()); see https://jb.gg/ij-platform-threading for details
Current thread: Thread[pool-4-thread-1,5,main] 1955244535 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1744828808
	at com.intellij.util.concurrency.ThreadingAssertions.createThreadAccessException(ThreadingAssertions.java:149)
	at com.intellij.util.concurrency.ThreadingAssertions.softAssertReadAccess(ThreadingAssertions.java:107)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1012)
	at com.sourcegraph.cody.ignore.IgnoreNotificationProvider$Companion.updateNotifications(IgnoreNotificationProvider.kt:49)
	at com.sourcegraph.cody.ignore.IgnoreOracle.onIgnoreDidChange(IgnoreOracle.kt:59)
	at com.sourcegraph.cody.agent.CodyAgentService$1.invoke$lambda$9(CodyAgentService.kt:84)
	at com.sourcegraph.cody.agent.CodyAgentClient.ignoreDidChange(CodyAgentClient.java:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)

```   

### Expected behavior

Cody should not crash and allow user to add accounts without any issues.

### Additional context

**Video link:**

https://drive.google.com/file/d/1wOKxQH2dyJakWdxpmkDGFjLAvqJ-MMiB/view?usp=drive_link

**Comment by taylorsperry** [comment]
Thanks, @jay-fibi. Are you able to repro this consistently?

**Comment by jay-fibi** [comment]
@taylorsperry yes, it is reproducible consistently and I could also record the video for the same.

**Comment by taylorsperry** [comment]
Thank you! Is this new in `5.5.9`? Can you repro in `5.5.8` and `5.5.5`? Sorry, I don't have a Windows machine and am trying to zero in on where this might have snuck in.

**Comment by taylorsperry** [comment]
cc @dominiccooney

**Comment by jay-fibi** [comment]
@taylorsperry I have checked and this is only reproducible on `5.5.9` and `5.5.8` only. I cannot reproduce it on `5.5.5`

**Comment by dominiccooney** [comment]
This is fixed in 4dc5732f0525229f021b8157327b4829c70aefd1, we just need to release that.

**Comment by dominiccooney** [comment]
And *that* has a defect @odisseus has fixed here: 0796b65f5c50374c28a2897d4eaf9ad4e9b08d41 so we need to roll past that at least.

**Comment by dominiccooney** [comment]
Should be fixed in 5.5.10.
